src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,141 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'react-peek/prop-types';
+import { lucidClassNames } from '../../util/style-helpers';
+import { buildHybridComponent } from '../../util/state-management';
+import {
+	ExpanderPanelDumb as ExpanderPanel,
+	IExpanderPanelHeaderProps,
+} from '../ExpanderPanel/ExpanderPanel';
+import {
+	findTypes,
+	omitProps,
+	StandardProps,
+	Overwrite,
+} from '../../util/component-types';
+import * as reducers from '../Accordion/Accordion.reducers';
+
+const cx = lucidClassNames.bind('&-Accordion');
+
+const { any, func, object, number, string } = PropTypes;
+
+interface IAccordionPropsRaw extends StandardProps {
+	/**
+	 * Indicates which item is expanded
+	 * */
+	selectedIndex: number;
+	/**
+	 * Called when the user clicks on the component's header of an item.
+	 * */
+
+	onSelect: (
+		selectedIndex: number | null,
+		{ event, props }: { event: React.MouseEvent; props: IAccordionProps }
+	) => void;
+	/**
+	 * Prop alternative to Header child component passed through to the
+	 * underlying ExpanderPanel
+	 */
+	Header: IExpanderPanelHeaderProps;
+}
+
+type IAccordionProps = Overwrite<
+	React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>,
+	IAccordionPropsRaw
+>;
+
+export interface IAccordionState {
+	selectedIndex: number | null;
+}
+
+const defaultProps = {
```

Comment:
So, looking at these default props when combined with `IAccordionPropsRaw` that means that `selectedIndex` and `Header` will be required but I don't think that's correct. There shouldn't be any required props.

Reply:
You're right, good catch. I noticed this when I converted examples to typescript, they did not typecheck.